### PR TITLE
Changes to allow ofxLua to work on latest OF

### PR DESCRIPTION
**Pull request by johnkingsley**
I'm not sure if you want these changes, but I thought I'd send them
in case you do want them -- it might save you some work.

These changes are the result of trying to get ofxLua working
on the latest github version of OF.

Many of the changes are related to the change in OF paths.
(std::filesystem::path is now used instead of std::string)

I can break these changes into smaller pull requests, if you prefer.

### CODE REVIEW

**Comment by danomatika** [comment]
As the readme states, the ofxLua master branch is kept up to date with the latest OF *release* and not current development work. We can wait for the next OF release or possible create a develop branch to incorporate this PR.

This is mainly because the API can change as development happens and it's simply easier for me to wait until things are finalized in a release.

File: openFrameworks.i [hunk 1]
```diff
@@ -1143,6 +1215,8 @@ class fstream {};
 
 // DIFF: ofURLFileLoader.h:ignoring ofHttpResponse ofBuffer operator
 %ignore ofHttpResponse::operator ofBuffer&();
+%ignore ofSaveURLTo(const string& url, const std::filesystem::path& path);
```
Comment (danomatika) [inline openFrameworks.i:1218]: Could keep these instead of ignoring, if possible. Just need filesystem to string conversion.

**Comment by danomatika** [comment]
Also, I will see if there is a way to use a typemap  swig can treat all std::filesystem::path types as std::string. As you've found, having to write manual wrappers to handle this is a pain.

File: openFrameworks.i (hunk 1 above)
Comment (johnkingsley) [reply openFrameworks.i:1218]: Yes, you're right, they should be kept.  In my particular case I didn't need these two methods.

**Comment by johnkingsley** [comment]
Yes, it makes sense to wait for the next official release.
I only made the pull request to maybe save someone the hassle of re-doing my work.
Unfortunately, the OF releases don't happen very often. :-(

It also makes sense to set up a typemap for std::filesystem::path <-> string conversion.
I thought about it, but my swig knowledge is very limited.

**Comment by danomatika** [comment]
I've incorporated your changes in the latest commits but it looks like there are quite a lot more extends to add. I messed around with a type map but didn't have too much luck, will need to try again.

Test and let me know if the current master builds for you with OF 0.10.0.

**Comment by johnkingsley** [comment]
Super! Thanks for the changes.

I tested it on the latest version of OF and created pull request #14 to fix some minor typos.

**Pull request closed without merging**